lightwallet-client-lib/src/main/java/co/electriccoin/lightwallet/client/internal/BlockingLightWalletClientImpl.kt

```diff
@@ -0,0 +1,191 @@
+package co.electriccoin.lightwallet.client.internal
+
+import cash.z.wallet.sdk.internal.rpc.CompactFormats
+import cash.z.wallet.sdk.internal.rpc.CompactTxStreamerGrpc
+import cash.z.wallet.sdk.internal.rpc.Service
+import co.electriccoin.lightwallet.client.BlockingLightWalletClient
+import co.electriccoin.lightwallet.client.ext.BenchmarkingExt
+import co.electriccoin.lightwallet.client.fixture.BlockRangeFixture
+import co.electriccoin.lightwallet.client.model.BlockHeightUnsafe
+import co.electriccoin.lightwallet.client.model.LightWalletEndpoint
+import co.electriccoin.lightwallet.client.model.LightWalletEndpointInfoUnsafe
+import co.electriccoin.lightwallet.client.model.RawTransactionUnsafe
+import co.electriccoin.lightwallet.client.model.Response
+import co.electriccoin.lightwallet.client.model.SendResponseUnsafe
+import com.google.protobuf.ByteString
+import io.grpc.Channel
+import io.grpc.ConnectivityState
+import io.grpc.ManagedChannel
+import io.grpc.StatusRuntimeException
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Implementation of LightwalletService using gRPC for requests to lightwalletd.
+ *
+ * @property channel the channel to use for communicating with the lightwalletd server.
+ * @property singleRequestTimeout the timeout to use for non-streaming requests. When a new stub
+ * is created, it will use a deadline that is after the given duration from now.
+ * @property streamingRequestTimeout the timeout to use for streaming requests. When a new stub
+ * is created for streaming requests, it will use a deadline that is after the given duration from
+ * now.
+ */
+internal class BlockingLightWalletClientImpl private constructor(
+    private val channelFactory: ChannelFactory,
+    private val lightWalletEndpoint: LightWalletEndpoint,
+    private val singleRequestTimeout: Duration = 10.seconds,
+    private val streamingRequestTimeout: Duration = 90.seconds
```

Comment:
Beware of timeouts being handled slightly different than one might expect where the call can be terminated because it hasn't finished its work in the duration specified regardless it is sending information back or not. This was a particular headache I suffered in the iOS version of GRPC